### PR TITLE
BUG: Fix BaseWindowGroupby.aggregate where as_index is ignored

### LINKED ISSUE #31007: `as_index = False` not working for applying groupby rolling agg to DataFrame
#### Problem description

When using a rolling agg function on a groupby object, we cannot ommit the groupby columns from the resulting dataframe's index using `as_index = False`, like we can when applying a non-rolling agg function.

#### Code Sample

```python
import pandas as pd

data = {
    'groupby_col': ['A', 'A', 'A', 'A', 'A', 'B', 'B', 'B', 'B', 'B', ],
    'agg_col': [1, 1, 0, 1, 0, 0, 0, 0, 1, 0],
}
df = pd.DataFrame(data)
df.groupby(['groupby_col'], as_index=False).rolling(4).agg({'agg_col': 'mean'})

```

#### Expected Output
```python
pd.DataFrame({'agg_col': {
  0: np.nan,
  1: np.nan,
  2: np.nan,
  3: 0.75,
  4: 0.5,
  5: np.nan,
  6: np.nan,
  7: np.nan,
  8: 0.25,
  9: 0.25}})
```

<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>agg_col</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>NaN</td>
    </tr>
    <tr>
      <th>1</th>
      <td>NaN</td>
    </tr>
    <tr>
      <th>2</th>
      <td>NaN</td>
    </tr>
    <tr>
      <th>3</th>
      <td>0.75</td>
    </tr>
    <tr>
      <th>4</th>
      <td>0.50</td>
    </tr>
    <tr>
      <th>5</th>
      <td>NaN</td>
    </tr>
    <tr>
      <th>6</th>
      <td>NaN</td>
    </tr>
    <tr>
      <th>7</th>
      <td>NaN</td>
    </tr>
    <tr>
      <th>8</th>
      <td>0.25</td>
    </tr>
    <tr>
      <th>9</th>
      <td>0.25</td>
    </tr>
  </tbody>
</table>

#### Actual Output
```python
pd.DataFrame({'agg_col': {
  (0, 0): np.nan,
  (0, 1): np.nan,
  (0, 2): np.nan,
  (0, 3): 0.75,
  (0, 4): 0.5,
  (1, 5): np.nan,
  (1, 6): np.nan,
  (1, 7): np.nan,
  (1, 8): 0.25,
  (1, 9): 0.25}})
```

<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th></th>
      <th>agg_col</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th rowspan="5" valign="top">0</th>
      <th>0</th>
      <td>NaN</td>
    </tr>
    <tr>
      <th>1</th>
      <td>NaN</td>
    </tr>
    <tr>
      <th>2</th>
      <td>NaN</td>
    </tr>
    <tr>
      <th>3</th>
      <td>0.75</td>
    </tr>
    <tr>
      <th>4</th>
      <td>0.50</td>
    </tr>
    <tr>
      <th rowspan="5" valign="top">1</th>
      <th>5</th>
      <td>NaN</td>
    </tr>
    <tr>
      <th>6</th>
      <td>NaN</td>
    </tr>
    <tr>
      <th>7</th>
      <td>NaN</td>
    </tr>
    <tr>
      <th>8</th>
      <td>0.25</td>
    </tr>
    <tr>
      <th>9</th>
      <td>0.25</td>
    </tr>
  </tbody>
</table>

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.7.3.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 18.7.0
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : en_US.UTF-8

pandas           : 0.25.1
numpy            : 1.17.4
pytz             : 2019.3
dateutil         : 2.8.0
pip              : 19.3.1
setuptools       : 40.8.0
Cython           : 0.29.14
pytest           : 5.3.2
hypothesis       : None
sphinx           : 2.3.0
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : 1.0.1
pymysql          : None
psycopg2         : 2.8.4 (dt dec pq3 ext lo64)
jinja2           : 2.10.3
IPython          : 7.10.2
pandas_datareader: None
bs4              : 4.8.1
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : None
matplotlib       : 3.1.2
numexpr          : 2.7.1
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 0.15.1
pytables         : None
s3fs             : None
scipy            : 1.3.0
sqlalchemy       : 1.3.11
tables           : 3.6.1
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None

</details>

### PR DESCRIPTION
**Pull request by ihsansecer**
- [x] closes #31007
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/v2.2.0.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/window/rolling.py [hunk 1]
```diff
@@ -875,6 +875,14 @@ def _gotitem(self, key, ndim, subset=None):
             subset = self.obj.set_index(self._on)
         return super()._gotitem(key, ndim, subset=subset)
 
+    def aggregate(self, func, *args, **kwargs):
+        result = super().aggregate(func, *args, **kwargs)
+        if not self._as_index:
```
Comment (ihsansecer) [inline pandas/core/window/rolling.py:880]: I think this should live in `BaseWindowGroupby` so added this aggregate function. After adding, mypy complained about incompatibility of base classes for classes inheriting `BaseWindowGroupby`. Adding aggregate function to those solves the issue happy to extend their docs if required or implement another workaround if preferred over this

File: pandas/core/window/ewm.py [hunk 2]
```diff
@@ -919,6 +919,12 @@ def _get_window_indexer(self) -> GroupbyIndexer:
         )
         return window_indexer
 
+    @doc(_shared_docs["aggregate"])
```
Comment (mroeschke) [inline pandas/core/window/ewm.py:922]: Shouldn't this and below be inherited from the base class?

File: doc/source/whatsnew/v2.2.0.rst [hunk 3]
```diff
@@ -168,6 +168,7 @@ Performance improvements
 Bug fixes
 ~~~~~~~~~
 - Bug in :class:`AbstractHolidayCalendar` where timezone data was not propagated when computing holiday observances (:issue:`54580`)
+- Bug in :meth:`BaseWindowGroupby.aggregate` where ``_as_index`` attribute is ignored (:issue:`31007`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.2.0.rst:171]: Could you write this in terms of public APIs?

File: pandas/core/window/ewm.py (hunk 2 above)
Comment (ihsansecer) [reply pandas/core/window/ewm.py]: I think so. This was an ugly workaround to a mypy error I was getting but using `type: ignore[misc]` instead now

File: doc/source/whatsnew/v2.2.0.rst (hunk 3 above)
Comment (ihsansecer) [reply doc/source/whatsnew/v2.2.0.rst]: Changed this now

**Comment by ihsansecer** [comment]
Also realized that this is failing with a list-like parameter as well so extended testing

**Review by rhshadrach — CHANGES_REQUESTED** [review]
I think this might need some more discussion before implementing; I've put some thoughts in the corresponding issue.

**Pull request closed without merging**